addons/stock/models/stock_move_line.py

```diff
@@ -510,15 +510,15 @@ def _free_reservation(self, product_id, location_id, quantity, lot_id=None, pack
                         candidate.product_uom_qty = 0.0
                     else:
                         candidate.unlink()
+                    if float_is_zero(quantity, precision_rounding=rounding):
+                        break
                 else:
                     # split this move line and assign the new part to our extra move
                     quantity_split = float_round(
                         candidate.product_qty - quantity,
                         precision_rounding=self.product_uom_id.rounding,
                         rounding_method='UP')
                     candidate.product_uom_qty = self.product_id.uom_id._compute_quantity(quantity_split, candidate.product_uom_id, rounding_method='HALF-UP')
-                    quantity -= quantity_split
                     move_to_recompute_state |= candidate.move_id
-                if quantity == 0.0:
                     break
```

Comment:
It seems that it is not wanted to break the loop here, right?

Reply:
Haha, ok, just saw the outdated comment of Simon, but not sure about the change anyway

Reply:
Thx, it sounds like you know exactly what you have done...so no concerns (as long as I won't get hit by it later) :wink: